termux-app/terminal-term/src/main/java/com/termux/app/TermuxActivity.java

```diff
@@ -299,8 +298,19 @@ public void onPageSelected(int position) {
         registerForContextMenu(mTerminalView);
 
         Intent serviceIntent = new Intent(this, TermuxService.class);
-        intentData = getIntent().getData();
-        serviceIntent.setData(intentData);
+        String intentData = getIntent().getDataString();
+        if (intentData == null || intentData.isEmpty()) {
+            showErrorAndGoBackToUserland("Intent data is empty.  Only UserLAnd intents are supported");
```

Comment:
This should use a string resource. If you can add it to the resources under UserLAnd and it can be shared with Termux that would be ideal.

Reply:
I'll add it as a string resource but keeping it in the module's string.xml resource file may be better.  It'll keep both projects decoupled and allow us to build both terminal and userland projects in parallel.